src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java

```diff
@@ -62,7 +62,7 @@ public static FORMAT valueOf(byte b) {
     /**
      * Force a max size on the fronting set to prevent uncontrolled growth and OOM errors
      */
-    public static final int MAX_FRONTING_SET_SIZE = 1000;
```

Comment:
The bug comes from combining the fronting set and the chunk list.

The bug can easily be removed by exclusively using one or the other:
 - `MAX_FRONTING_SET_SIZE = 0` to solely use the chunk list
 - `MAX_FRONTING_SET_SIZE = INFINITY` to solely use the fronting set

But it would be best to still use both and find where the bug is.